src/main/java/com/SAS/User/Privileges.java

```diff
@@ -1,13 +1,76 @@
 /**
- * The class represents the privileges of registered user
+ * The class represents the a privileges singleton that will be held by the role class
  */
 
 package com.SAS.User;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
 
 public class Privileges {
 
-    //Will be implemented
-    public void edit(String privilege) throws Exception {
-        throw new Exception("Not implemented");
+    private volatile static Privileges instance = null;
+    public HashMap<String, HashSet<String>> privileges;
+
+    /**
+     * constructor for the singleton
+     */
+    private Privileges(){
+        privileges = new HashMap<>();
+        initPrivileges();
+    }
+
+    public static Privileges getInstance() {
+        if(instance == null){
+            synchronized(Privileges.class){
+                if(instance == null) {
+                    instance = new Privileges();
+                }
+            }
+        }
+        return instance;
+    }
+
+    // <editor-fold desc="Initialize privileges HashMap"
+    private void initPrivileges() {
+        this.privileges.put("Guest", new HashSet<>(Arrays.asList
+                ("viewPages+LS", "searchByNCK")));
+        this.privileges.put("Fan", new HashSet<>(Arrays.asList
+                ("followPage", "getNoti", "fileReport", "viewSHistory", "editPDetails")));
+        this.privileges.put("Player", new HashSet<>(Arrays.asList
+                ("editPDetails", "uploadPContent")));
+        this.privileges.put("Coach", new HashSet<>(Arrays.asList
+                ("editPDetails", "uploadPContent")));
+        this.privileges.put("TeamOwner", new HashSet<>(Arrays.asList
+                ("add/removeA", "add/removeTO", "add/removeTM", "editTMPriv", "closeTNP", "addTrans", "editPDetails")));
+        this.privileges.put("TeamManager", new HashSet<>(Arrays.asList
+                ("add/removeA", "editPDetails")));
+        this.privileges.put("SystemAdmin", new HashSet<>(Arrays.asList
+                ("editPDetails", "closeTP", "removeRegistered", "view/replyReport", "viewSysSettings", "enableRS")));
+        this.privileges.put("AssociationRepresentative", new HashSet<>(Arrays.asList
+                ("editPDetails", "defineL", "defineSL", "add/removeR", "setRinL", "define/changePolicy", "enableGScheduling", "defineBudgetRules", "manageAssBudget")));
+        this.privileges.put("Referee", new HashSet<>(Arrays.asList
+                ("editPDetails", "viewMyGames","editGameEvent")));
+    }
+    // </editor-fold>
+
+    /**
+     * This function returns the appropriate privileges for the received role
+     * @param privilege - a string representing the role
```

Comment:
Maybe consider change "privilege" param name to "role" or something like that.
for me, at first, it was unclear.

Reply:
We'll change it.